shared/eval/src/harness.rs

```diff
@@ -674,6 +677,22 @@ impl PreparedTask {
             if generation_complete {
                 cache.write().unwrap().remove(&doc_index);
 
+                // Extract answer from the complete generated text using regex
+                if let Ok(generated_text) = tokenizer.decode(&generated_tokens, false) {
+                    if let Some(captures) = answer_extraction_regex.captures(&generated_text) {
```

Comment:
What happens here if the regex contains more than one match?, We should ensure that we are getting only the last one, or only make the regex with the last part of the generated text